### PR TITLE
Postgres default Engine upgrade

### PR DESCRIPTION
**Pull request by laurafitzgerald**
## Overview

This is WIP and needs some cleanup and test updates

## Verification

- Check out master
- Run `make cluster/prepare`
- Run `make run`
- Run `make cluster/seed/managed/postgres`
- verify postgres is engine version 1.15 and that no service update is showing in the UI
- Checkout this branch
- Rerun the code
- Add applyImmediately true to the postgres cr to trigger the engine upgrade
- Once the engine upgrade is present see that the system update is available
- the rest is todo but the idea is that the service update gets applied immediately after the engine upgrade.

## Checklist
- [ ] This PR includes a change to an instance type, I have used script `hack/<provider>/supported_types.sh` and attached the result below

### CODE REVIEW

File: pkg/providers/aws/provider_postgres.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ const (
 	defaultAwsBackupRetentionPeriod      = 31
 	defaultAwsDBInstanceClass            = "db.t3.small"
 	defaultAwsEngine                     = "postgres"
-	defaultAwsEngineVersion              = "10.15"
+	defaultAwsEngineVersion              = "10.16"
```
Comment (laurafitzgerald) [inline pkg/providers/aws/provider_postgres.go:55]: We can probably take this to a higher value

File: pkg/providers/aws/provider_postgres.go [hunk 2]
```diff
@@ -211,8 +211,46 @@ func (p *PostgresProvider) CreatePostgres(ctx context.Context, pg *v1alpha1.Post
...
+	session := rds.New(sess)
 	// create the aws RDS instance
-	return p.createRDSInstance(ctx, pg, rds.New(sess), ec2.New(sess), rdsCfg, isEnabled)
+	postgres, status, err := p.createRDSInstance(ctx, pg, session, ec2.New(sess), rdsCfg, isEnabled)
+	if err != nil {
+		errMsg := "failed to create rds instance"
+		return nil, status, errorUtil.Wrap(err, errMsg)
+	}
+
+	// Retrieve service maintenance updates, create and export Prometheus metrics
```
Comment (laurafitzgerald) [inline pkg/providers/aws/provider_postgres.go:222]: Ideally this block can go into it's own function or at the bottom of the createRDSInstance function if it works there.

File: pkg/providers/aws/provider_postgres.go [hunk 3]
```diff
@@ -211,8 +211,46 @@ func (p *PostgresProvider) CreatePostgres(ctx context.Context, pg *v1alpha1.Post
...
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to get pending maintenance information for RDS with identifier : %s", rdsCfg.DBInstanceIdentifier)
+		return nil, croType.StatusMessage(errMsg), errorUtil.Wrap(err, errMsg)
+	}
+	for _, rpma := range output.PendingMaintenanceActions {
+		// todo if the resource identifier does not equal the arn of the instance
+		/*if !rpma.ResourceIdentifier
+			continue
+		}*/
+		for _, pmac := range rpma.PendingMaintenanceActionDetails {
```
Comment (laurafitzgerald) [inline pkg/providers/aws/provider_postgres.go:233]: I haven't tested this loop yet or the ApplyPendingMaintenanceAction function yet.

**Comment by austincunningham** [comment]
## Verification: 
- installed master  
- rds engine version 10.15
image
- no pending updates
```bash
INFO[1107] checking for pending postgres service updates 
INFO[1108] rds serviceupdates: 0 available  
```
- no pending maintenance in the UI
image

- checked out this branch MGDAPI-3095
- `make run`
Seeing an error in postgres CR status and in the logs
```yaml
status:
  message: "failed to get pending maintenance information for RDS with identifier : mwcollabbyoc8l5nbcloudresourceopera-yxkj: failed to get pending maintenance information for RDS with identifier : mwcollabbyoc8l5nbcloudresourceopera-yxkj: InvalidParameterValue: Invalid resource identifier:  mwcollabbyoc8l5nbcloudresourceopera-yxkj\n\tstatus code: 400, request id: 3e2389a7-5b5d-4d63-89e3-12ea935c181b"
```
- added applyImmediately: true to the spec of the postgres CR
```yaml
spec:
  secretRef:
    name: example-postgres-sec
  tier: development
  type: managed
  applyImmediately: true
```

- error continues after CR update. looks to be on the beginning of the  DescribePendingMaintenanceActions function  
https://github.com/integr8ly/cloud-resource-operator/pull/235/commits/59be2545b8a61556104162e46d45c20e8b521a3b#

**Comment by MStokluska** [comment]
I've reviewed the functionality on @laurafitzgerald cluster, we have run the upgrade scenario and it all looks good. Code looks good as well. 
Good job!
/lgtm
/approve

**Pull request merged**